### PR TITLE
Fixed door unlocking after calling Door.Lock multiple times with the same DoorLockType

### PR DESCRIPTION
**Pull request by JustMarfix**
## Description
Fixed door unlocking after calling the Door.Lock multiple times with the same DoorLockType.

**What is the current behavior?** (You can also link to an open issue here)
The Door.Lock method unlocks the door when the same DoorLockType has been used multiple times due to the code in Door.ChangeLock. It doesn't match the method name, so I fixed it.

**What is the new behavior?** (if this is a feature change)
Now, the ChangeLock method only can add new locks.

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

**Other information**:
-
<br />

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentations
<br />

## Submission checklist

- [x] I have checked the project can be compiled
- [x] I have tested my changes and it worked as expected

### Patches (if there are any changes related to Harmony patches)
- [ ] I have checked no IL patching errors in the console

### Other
- [ ] Still requires more testing

### CODE REVIEW

**Review by VladTheCow — DISMISSED** [review]
PRs should target the dev branch, not master

**Comment by Mikihero** [comment]
https://discord.com/channels/656673194693885975/656674085425643534/1305208899288436779

tl;dr `Door::Lock` is named a bit confusingly for what it does, changes should be made to the `Lock` method directly and not `ChangeLock`

**Review by louis1706 — APPROVED** [review]

**Comment by VALERA771** [comment]
Will your fix display active locks correctly?

**Comment by louis1706** [comment]
> Will your fix display active locks correctly?

DoorEvents.TriggerAction that what it does

**Comment by louis1706** [comment]
i have an suggestion can you add Lock method without the time argument

**Comment by JustMarfix** [comment]
> i have an suggestion can you add Lock method without the time argument

done

**Review by louis1706 — APPROVED** [review]

**Review by VALERA771 — APPROVED** [review]

**Review by Misfiy — APPROVED** [review]

**Review by louis1706 — APPROVED** [review]
